### PR TITLE
Re-add ingest project

### PR DESCRIPTION
**Pull request by jpolchlo**
We lost the ingest project in the transition to vector pipe, since it would require a rewrite to accomodate the new vectortile creation pipeline.  This PR reinstates that project and populates it with a pair of utilities.  

QA tile generator:Pulls a sampling of features from an OSM snapshot, and places each type in a different layer of a vector tile.

Footprint rasterizer: Generates ZXY layer of PNG tiles based on custom colorization rules specified by a JSON configuration.

Closes azavea/geotrellis#127
Closes azavea/geotrellis#126

Signed-off-by: jpolchlo <jpolchlopek@azavea.com>

### CODE REVIEW

File: src/ingest/build.sbt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  sparkJts,
+  gtShapefile exclude("com.google.protobuf", "protobuf-java"),
+  "com.google.protobuf" % "protobuf-java" % "2.5.0",
+  vectorpipe exclude("com.google.protobuf", "protobuf-java"),
+  gtRaster,
+  gtS3,
+  cats,
+  scalactic,
+  scalatest,
+  "com.amazonaws" % "aws-java-sdk-s3" % "1.11.340" % "provided"
```
Comment (mojodna) [inline src/ingest/build.sbt]: Maybe incorporate this approach instead? https://github.com/azavea/osmesa/issues/159

(or at least see if it's relevant here; EMR compatibility was the initial reason for setting `provided`. I also seem to remember having to do some digging to figure out version numbers for AWS-related dependencies, which excluding by wildcard would simplify tremendously.)

File: src/ingest/src/main/scala/osmesa/ingest/QATiles.scala [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    val maxZoomOpt = Opts.option[Int]("maxZoom", help = "Largest (most resolute) zoom level to generate").withDefault(15)
+    val aoiOpt = Opts.option[URI]("aois", help = "Location of GeoJSON file giving areas of interest").withDefault(new URI(""))
+      .validate("AOI file must be a .geojson file") { uri => uri.toString.isEmpty || uri.toString.endsWith(".geojson") }
+    val orcArg = Opts
+      .argument[URI]("source ORC file")//, help = "Location of the ORC file containing geometries to process")
+      .validate("URI to ORC must have an s3 or file scheme") { _.getScheme != null }
+      .validate("orc must be an S3 or file Uri") { uri =>
+        uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+      }
+      .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
```
Comment (mojodna) [inline src/ingest/src/main/scala/osmesa/ingest/QATiles.scala]: When writing outputs with multiple partitions, it puts individual ORC files into a directory. Unless we want to enforce that the containing directory include `.orc`, let's drop this.

File: src/ingest/src/main/scala/osmesa/ingest/QATiles.scala [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+      val pipeline = QATilesPipeline("geom", outputUri)
+
+      VectorPipe(features, pipeline, VectorPipe.Options.forZoomRange(minZoom, maxZoom))
+    }
+  }
+)
+
+object QAFunctions {
+  val st_emptyGeom = functions.udf { g: Geometry => g.isEmpty }
```
Comment (mojodna) [inline src/ingest/src/main/scala/osmesa/ingest/QATiles.scala]: [`st_isEmpty`](https://www.geomesa.org/documentation/user/spark/sparksql_functions.html#st-isempty)?

File: src/ingest/src/main/scala/osmesa/ingest/RasterFootprints.scala [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+
+import java.io.ByteArrayInputStream
+import java.net.URI
+import java.util.concurrent.Executors
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.ExecutionContext
+
+object RasterizeOSM extends CommandApp(
+  name="qa-tile-generator",
+  header=""" Create QA tiles from OSM ORC file
```
Comment (mojodna) [inline src/ingest/src/main/scala/osmesa/ingest/RasterFootprints.scala]: rasterized, yeah?

File: src/ingest/src/main/scala/osmesa/ingest/RasterFootprints.scala [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+    val maxZoomOpt = Opts.option[Int]("maxZoom", help = "Largest (most resolute) zoom level to generate").withDefault(15)
+    val tileResOpt = Opts.option[Int]("resolution", help = "Resolution of output tile in pixels").withDefault(256)
+
+    val orcArg = Opts
+      .argument[URI]("source ORC file")
+      .validate("URI to ORC must have an s3 or file scheme") { _.getScheme != null }
+      .validate("orc must be an S3 or file Uri") { uri =>
+        uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+      }
+      .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
```
Comment (mojodna) [inline src/ingest/src/main/scala/osmesa/ingest/RasterFootprints.scala]: Same as above w/r/t directories containing ORC files.

**Comment by echeipesh** [comment]
Alas, perhaps in a better world.

**Pull request closed without merging**